test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -103,381 +108,665 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
+					gomega.Expect(func() error {
+						w := createdWorkload.DeepCopy()
+						workload.SetEvictedCondition(w, "ByTest", "by test")
+						return workload.ApplyAdmissionStatus(ctx, k8sClient, w, false)
+					}()).Should(gomega.Succeed())
+
+					gomega.Consistently(func(g gomega.Gomega) {
+						g.Expect(k8sClient.Get(ctx, pod1LookupKey, createdPod1)).To(gomega.Succeed())
+						g.Expect(k8sClient.Get(ctx, pod2LookupKey, createdPod2)).To(gomega.Succeed())
+					}, util.ConsistentDuration, util.Interval).Should(gomega.Succeed())
 				})
 
-				ginkgo.By("add labels & annotations to the admission check", func() {
-					gomega.Eventually(func() error {
-						var newWL kueue.Workload
-						gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(createdWorkload), &newWL)).To(gomega.Succeed())
-						workload.SetAdmissionCheckState(&newWL.Status.AdmissionChecks, kueue.AdmissionCheckState{
-							Name:  "check",
-							State: kueue.CheckStateReady,
-							PodSetUpdates: []kueue.PodSetUpdate{
-								{
-									Name: "main",
-									Labels: map[string]string{
-										"label1": "label-value1",
-									},
-									Annotations: map[string]string{
-										"ann1": "ann-value1",
-									},
-									NodeSelector: map[string]string{
-										"selector1": "selector-value1",
-									},
-								},
-							},
-						})
-						return k8sClient.Status().Update(ctx, &newWL)
+				ginkgo.By("finish one pod and fail the other, the eviction should end", func() {
+					util.SetPodsPhase(ctx, k8sClient, corev1.PodSucceeded, pod1)
+					util.SetPodsPhase(ctx, k8sClient, corev1.PodFailed, pod2)
+
+					gomega.Eventually(func(g gomega.Gomega) {
+						g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+						g.Expect(createdWorkload.Status.Conditions).Should(gomega.ContainElement(
+							gomega.BeComparableTo(metav1.Condition{Type: kueue.WorkloadAdmitted, Status: metav1.ConditionFalse}, wlConditionCmpOpts...),
+						))
 					}, util.Timeout, util.Interval).Should(gomega.Succeed())
 				})
 
-				ginkgo.By("admit the workload", func() {
-					admission := testing.MakeAdmission(clusterQueueAc.Name).
-						Assignment(corev1.ResourceCPU, "test-flavor", "1").
-						AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
-						Obj()
-					gomega.Expect(k8sClient.Get(ctx, *wlLookupKey, createdWorkload)).Should(gomega.Succeed())
+				replacementPod := testingpod.MakePod("test-pod2-replace", ns.Name).
+					Group("test-group").
+					GroupTotalCount("2").
+					Queue("test-queue").
+					Obj()
+				replacementPodLookupKey := client.ObjectKeyFromObject(pod2)
+
+				ginkgo.By("creating the replacement pod and readmitting the workload will unsuspended the replacement", func() {
+					gomega.Expect(k8sClient.Create(ctx, replacementPod)).Should(gomega.Succeed())
+
+					gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
```

Comment:
worth checking that this is the same Workload that was initially created. The UID should match